File: Searching/Linear_Search/C/LinearSearch.c
#include "stdio.h"

int linearSearch(int itemList[],int itemListSize,int item)
{
    int index=0;
    while (index < itemListSize)
    {
        if (itemList[index] == item)
        {
            return index;
        }
        index++;
    }
    return -1;
    
}
 //Simple demonstration for the algorithm
int main(int argc,char* argv[])
{   
    int arr[] = {1,2,3,4,5};
    printf("%d\n",linearSearch(arr,sizeof(arr)/sizeof(arr[0]),3));
    printf("%d",linearSearch(arr,sizeof(arr)/sizeof(arr[0]),0));
    
}

File: Searching/Linear_Search/Python/LinearSearch.py
"""
This is a simple Linear search algorithm for python.
This is the exact implementation of the pseudocode written in README.
"""
def linear_search(item_list,item):
    index=0
    while index < len(item_list):
        if item_list[index] == item:
            return index
        index+=1
    return None

#Simple demonstration of the function
if __name__=='__main__':
    print (linear_search([1,3,5,7],3))
    print (linear_search([1,2,5,7,97],0))
